File: app/src/main/java/org/ametro/model/serialization/GlobalIdentifierProvider.java
package org.ametro.model.serialization;

public class GlobalIdentifierProvider {

    private int segmentCounter = 0x10000000;
    private int transferCounter = 0x20000000;

    public int getSegmentUid() {
        return segmentCounter++;
    }
    public int getTransferUid() {
        return transferCounter++;
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/model/serialization/GlobalIdentifierProvider.kt
package io.github.romangolovanov.apps.ametro.model.serialization

class GlobalIdentifierProvider {
    private var segmentCounter = 0x10000000
    private var transferCounter = 0x20000000

    fun getSegmentUid(): Int = segmentCounter++
    fun getTransferUid(): Int = transferCounter++
}
